mcux/middleware/wifi_nxp/CMakeLists.txt

```diff
@@ -24,120 +24,132 @@ zephyr_compile_definitions_ifdef(CONFIG_NXP_RW610
     WIFI_BT_TX_PWR_LIMITS="wlan_txpwrlimit_cfg_WW_rw610.h"
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_88W8987_AW_CM358_USD
+zephyr_compile_definitions_ifdef(CONFIG_NXP_88W8987_AW_CM358_USD
     WIFI_BT_TX_PWR_LIMITS="wlan_txpwrlimit_cfg_WW.h"
     CONFIG_NXP_WIFI_ED_OFFSET_2G=0x09
     CONFIG_NXP_WIFI_ED_OFFSET_5G=0x0C
     WIFI_BT_USE_USD_INTERFACE
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_88W8987_AW_CM358MA_M2
+zephyr_compile_definitions_ifdef(CONFIG_NXP_88W8987_AW_CM358MA_M2
     WIFI_BT_TX_PWR_LIMITS="wlan_txpwrlimit_cfg_WW.h"
     CONFIG_NXP_WIFI_ED_OFFSET_2G=0x09
     CONFIG_NXP_WIFI_ED_OFFSET_5G=0x0C
     WIFI_BT_USE_M2_INTERFACE
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_88W8987_MURATA_1ZM_USD
+zephyr_compile_definitions_ifdef(CONFIG_NXP_88W8987_MURATA_1ZM_USD
     WIFI_BT_TX_PWR_LIMITS="wlan_txpwrlimit_cfg_murata_1ZM_WW.h"
     CONFIG_NXP_WIFI_ED_OFFSET_2G=0x06
     CONFIG_NXP_WIFI_ED_OFFSET_5G=0x06
     WIFI_BT_USE_USD_INTERFACE
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_88W8987_MURATA_1ZM_M2
+zephyr_compile_definitions_ifdef(CONFIG_NXP_88W8987_MURATA_1ZM_M2
     WIFI_BT_TX_PWR_LIMITS="wlan_txpwrlimit_cfg_murata_1ZM_WW.h"
     CONFIG_NXP_WIFI_ED_OFFSET_2G=0x06
     CONFIG_NXP_WIFI_ED_OFFSET_5G=0x06
     WIFI_BT_USE_M2_INTERFACE
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_IW416_AW_AM457_USD
+if(CONFIG_NXP_88W8987)
```

Comment:
This approach makes sense, but we should expand it out to manage other modules (for example the IW416)